webapp-django/crashstats/manage/views.py

```diff
@@ -789,3 +792,140 @@ def _get_edit_url(action, extra):
+    for permission in all_possible_permissions:
+        possible_permissions.append(permission)
+
+    expires_choices = (
+        (1, '1 day'),
+        (7, '1 week'),
+        (30, '1 month'),
+        (30 * 3, '3 months'),
+        (365, '1 year'),
+        (365 * 10, '10 years'),
+    )
+
+    if request.method == 'POST':
+        form = forms.APITokenForm(
+            request.POST,
+            possible_permissions=possible_permissions,
+            expires_choices=expires_choices,
+        )
+        if form.is_valid():
+            data = form.cleaned_data
+            token = Token.objects.create(
+                user=data['user'],
+                notes=data['notes'],
+                expires=data['expires'],
+            )
+            for permission in data['permissions']:
+                token.permissions.add(permission)
+
+            messages.success(
+                request,
+                'API Token for %s created. ' % token.user.email
+            )
+            return redirect('manage:api_tokens')
+    else:
+        form = forms.APITokenForm(
+            possible_permissions=possible_permissions,
+            expires_choices=expires_choices,
+            initial={
+                'expires': settings.TOKENS_DEFAULT_EXPIRATION_DAYS,
+            }
+        )
+    context = {
+        'form': form,
+    }
+    return render(request, 'manage/api_tokens.html', context)
+
+
+@require_POST
+@json_view
+@superuser_required
+def api_tokens_delete(request):
+    if not request.POST.get('id'):
+        return http.HttpResponseBadRequest('No id')
+    token = get_object_or_404(Token, id=request.POST['id'])
+    token.delete()
+    return True
+
+
+@json_view
+@superuser_required
+def api_tokens_data(request):
+    form = forms.FilterAPITokensForm(request.GET)
+    if not form.is_valid():
+        return http.HttpResponseBadRequest(str(form.errors))
+
+    tokens = Token.objects.all().order_by('-created')
+    if form.cleaned_data['user']:
+        tokens = tokens.filter(
+            user__email__icontains=form.cleaned_data['user']
+        )
+    if form.cleaned_data['key']:
+        tokens = tokens.filter(
+            key__startswith=form.cleaned_data['key']
+        )
+    if form.cleaned_data['expired'] == 'yes':
+        tokens = tokens.filter(
+            expires__lt=timezone.now()
+        )
+    elif form.cleaned_data['expired'] == 'no':
+        tokens = tokens.filter(
+            expires__gte=timezone.now()
+        )
+    count = tokens.count()
+    try:
+        page = int(request.GET.get('page', 1))
+        assert page >= 1
+    except (ValueError, AssertionError):
+        return http.HttpResponseBadRequest('invalid page')
+
+    items = []
+    batch_size = settings.API_TOKENS_ADMIN_BATCH_SIZE
+    batch = Paginator(tokens.select_related('user'), batch_size)
+    batch_page = batch.page(page)
+
+    # build up a dict of permission id -> permission name
+    _permissions_names = {}
```

Comment:
I guess it's a peterism. I use underscore to signify that the variable is only relevant in a temporary manner. It's not a variable that I want to keep and return as a context variable. 
Do you think it's ugly?